.github/workflows/pythonpublish.yml

```diff
@@ -1,24 +1,57 @@
 name: Upload Python Package
 
+# This is an adaptation of https://github.com/code-specialist/pypi-poetry-publish.
+# That action is released under an MIT License, and as such, so is this workflow.
+
 on:
   release:
-    types: [created]
+    types: [released]
 
 jobs:
   deploy:
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+
+      - name: Install poetry
+        run: |
+          pipx install poetry==${{ POETRY_VERSION }}
+
       - name: Set up Python
-        uses: actions/setup-python@v3
+        uses: actions/setup-python@v4
         with:
           python-version: "3.x"
+
+      - name: Set GitHub Tag as Package Version
+        run: |
+          sed -i -r 's/__version__ *= *".*"/__version__ = "${{ github.event.release.tag_name }}"/g' ./src/isar/__init__.py
+          sed -i '0,/version =.*/s//version = "'"${{ github.event.release.tag_name }}"'"/' ./pyproject.toml
+        shell: bash
```

Comment:
If we stick to setuptools, we can use the SCM tag (github release tag) for setting the version and we avoid having a separate version bump commit.

See example here: https://github.com/equinor/isar-robot/pull/103/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R34-R35